### PR TITLE
ICDS Dashboard: SDD pagination bug fix

### PR DESCRIPTION
**Pull request by saiprakash-v**
##### SUMMARY

This issue is raised due to persisting of SDD table state while switching between main table and details table. 

**Issue:** When table has 4 pages of entries, and moved to 4th page. Now, when the location is changed, and the new location has lesser number of entries/data but the table still remains in 4th page as shown in the image

**Initially when switched to 4th page**:
Screenshot from 2020-07-01 20-27-24

**after changing location, still remains in 4th page**:
Screenshot from 2020-07-01 20-28-29

In this PR I'm moving to first page as soon as there is a change in location
Before fix:
Screenshot from 2020-07-17 18-05-02
Screenshot from 2020-07-17 18-08-46
After fix:
Screenshot from 2020-07-17 18-12-26

### CODE REVIEW

**Comment by saiprakash-v** [comment]
Tagging @ShubhamBansal1997 to take an initial pass

**Review by ShubhamBansal1997 — APPROVED** [review]
Looks good to me(left one question), but it's good to test on staging as well.

File: custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js [hunk 1]
```diff
@@ -108,6 +108,31 @@ function ServiceDeliveryDashboardController($rootScope, $scope, $http, $location
...
+            var currentLocationId = $location.search()['location_id'];
+            if (currentLocationId !== storageService.getKey('currentLocationId')) {
+                storageService.setKey('currentLocationId', currentLocationId);
+                if (vm.step === 'pw_lw_children') {
+                    $('#sddPwLwChildrenTable').DataTable().page(0).draw(false);
+                    storageService.setKey('resetChildrenTable', true);
+                } else if (vm.step === 'children') {
+                    $('#sddChildrenTable').DataTable().page(0).draw(false);
+                    storageService.setKey('resetPwLwChildrenTable', true);
+                } else if (vm.step !== 'pw_lw_children' && vm.step !== 'children') {
```
Comment (ShubhamBansal1997) [inline custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js:131]: Does a simple else wouldn't work here or are you planning to add more else if clauses here??

Comment (saiprakash-v) [reply custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js:131]: Yeah. Simple else would work.

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
Why does in the second screenshot. Page numbers are not coming. I only see '1'. where are other page numbers? 
It says `showing 31 to 2 or 2 entries` which sounds confusing and indicates some bug. Could you explain is that telling

**Comment by saiprakash-v** [comment]
> Why does in the second screenshot. Page numbers are not coming. I only see '1'. where are other page numbers?
> It says `showing 31 to 2 or 2 entries` which sounds confusing and indicates some bug. Could you explain is that telling

First screenshot is at National level. So we had 31 entries (states). We navigated to 4th page. And then we selected a state (State poshan 1) which has only 1 district. Now ideally table should be reset to 1st page since we have only 1 entry. But the state of the table is persisting and still remains in the 4th page. That is the reason for "31" being displayed. This PR will reset the page number to 1, whenever there is a change in location id

**Review by Rohit25negi — APPROVED** [review]

**Pull request merged**